src/Framework/Telemetry/BuildTelemetry.cs

```diff
@@ -167,5 +167,56 @@ public override IDictionary<string, string> GetProperties()
 
             return properties;
         }
+
+        /// <summary>
+        /// Create a list of properties sent to VS telemetry with the information whether they should be hashed.
+        /// </summary>
+        /// <returns></returns>
+        public IList<TelemetryItem> GetActivityProperties()
+        {
+            List<TelemetryItem> telemetryItems = new(8);
+
+            if (StartAt.HasValue && FinishedAt.HasValue)
+            {
+                telemetryItems.Add(new TelemetryItem("BuildDurationInMilliseconds", (FinishedAt.Value - StartAt.Value).TotalMilliseconds, false));
+            }
+
+            if (InnerStartAt.HasValue && FinishedAt.HasValue)
+            {
+                telemetryItems.Add(new TelemetryItem("InnerBuildDurationInMilliseconds", (FinishedAt.Value - InnerStartAt.Value).TotalMilliseconds, false));
+            }
+
+            if (Host != null)
+            {
+                telemetryItems.Add(new TelemetryItem("BuildEngineHost", Host, false));
+            }
+
+            if (Success.HasValue)
+            {
+                telemetryItems.Add(new TelemetryItem("BuildSuccess", Success, false));
+            }
+
+            if (Target != null)
+            {
+                telemetryItems.Add(new TelemetryItem("BuildTarget", Target, true));
+            }
+
+            if (Version != null)
```

Comment:
can Version or other strings be empty?

Reply:
some of these string are sometimes empty, this is copied logic from sdk telemetry, I'm leaning towards that it's out of scope to refactor here